### DISCUSSION TITLE
C4GT: Auto generate content in local language

### DISCUSSION THREAD

**Discussion by cchaitanya27** [Sunbird-Ed/Community#489] [Feature request]
### Project Details
Sunbird ED is a platform that can be used to deploy scale-ready solutions in the area of learning, capacity building, and professional development. The platform provides capabilities for managing a variety of content in various languages.

This is to provide capability of translation of content from one English or an Indian language to other Indian languages using AI.  This will greatly help content creators to utilize existing content and repurpose it in their one language.

### Features to be implemented
A UI plugin that integrates with the ECML and Upload Editors of Sunbird ED.
The plugin has UI to 
Input textual content either through typing in the text, upload a pdf file or refer to an existing content on the platform.
Input source and target languages
The plugin takes the input, translates the text and provides the translated text. 
When the plugin is integrated with ECML Editor, the translated text appears as part of the slide in the Editor
When the plugin is integrated with the upload editor, it provides the translated text as a pdf.

### What exists
Sunbird ED Editors
Bhashini Services for Translation

### What needs to be built
Plugin that can be integrated with Sunbird ED Editors.

### Learning Path
Complexity: Low
Skills Required: 
Understanding existing Sunbird ED Editors, Bhashini, NodeJS, Angular
Mentors: Niharika, Chinmay
Project size: 4 weeks

### Milestones
Detailing and documenting the capabilities
Creating the design of the plugin UI
Implementation of the plugin
Test cases

**Comment by tejashg** [comment]
I am interested in working in the project, can you add Sunbird ED Editors and Bhashini Services for Translation repo links.

**Comment by chitranshu-keshav** [comment]
@Sunbird-Ed/sunbirded_workinggroup Can you please review and provide your inputs on this feature request

**Comment by tejashg** [comment]
@chitranshu-keshav  Hi, can you provide some editorials and github links for the project.

**Reply by tejashg** [reply]
Also, to which sunbird editor, we have to build plugin, for Interactive, file upload or collection editor?

**Reply by kameshbhr** [reply]
@tejashg Can you please join the Discord channel provided here: https://www.codeforgovtech.in/ and raise your queries there.

**Comment by chitranshu-keshav** [comment]
@Krishnaj20 can you please tag the relevant person to provide the above requested details

**Comment by Neha-sharma-13** [comment]
Hi Team,
I am confident that my technical skills, attention to detail, and ability to deliver high-quality results make me well-suited to take on this project. I am eager to collaborate with the team, contribute my expertise, and ensure a seamless integration of the translation plugin into Sunbird ED.

I kindly request that you consider assigning this issue to me as a contributor. I am committed to actively participating in the project, following best practices, and delivering timely and efficient solutions. 

Thank you for considering my request. I look forward to the opportunity to contribute to this impactful project.
